src/plugins/plot/inspector/forms/YAxisForm.vue

```diff
@@ -178,12 +178,6 @@ export default {
                         if (Number(range.min) > Number(range.max)) {
                             return 'Minimum must be less than Maximum.';
                         }
-
-                        if (model.get('autoscale')) {
-                            return false;
-                        }
-
-                        return true;
```

Comment:
not anymore. Now it returns a union of possible string values or undefined. THen further below it is used like this:

```js
            const validationError = formField.validate?.(newVal, this.yAxis);
```

`validationError` can be `undefined` (falsey) if `formField.validate` doesn't exist (the `?.` operator will return `undefined` in that case, or if the `validate` function returns `undefined`. Then, the code right after that does

```js
            if (validationError) {
                return;
```

in case there there is an error message, otherwise if no error (`undefined`) then it will continue.

Try it out in VS Code, it shows the type awareness (on hover, etc). There's also TypeScript plugins for pretty much all editors, but I haven't tried other ones. F.e. [WebStorm TypeScript](https://www.jetbrains.com/help/webstorm/typescript-support.html), totally worth it for web projects even with plain JS (it infers types in many cases without writing TS syntax, and can thus provide intellisense).

Reply:
Note how we no longer need to call the imported `validate()` function, because we cover that case with the new `?.` syntax.

Reply:
The operator is called [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)